pkg/controller/sesv2/configurationset/setup.go

```diff
@@ -0,0 +1,288 @@
+)
+
+const (
+	errNotConfigurationSet = "managed resource is not a SES ConfigurationSet custom resource"
+	errKubeUpdateFailed    = "cannot update SES ConfigurationSet custom resource"
+)
+
+// SetupConfigurationSet adds a controller that reconciles SES ConfigurationSet.
+func SetupConfigurationSet(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(svcapitypes.ConfigurationSetGroupKind)
+	opts := []option{setupExternal}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&svcapitypes.ConfigurationSet{}).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewController(rl),
+		}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.ConfigurationSetGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient()), managed.NewNameAsExternalName(mgr.GetClient()), &tagger{kube: mgr.GetClient()}),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func setupExternal(e *external) {
+	h := &hooks{client: e.client, kube: e.kube}
+	e.isUpToDate = isUpToDate
+	e.postObserve = postObserve
+	e.update = h.update
+}
+
+type hooks struct {
+	client sesv2iface.SESV2API
+	kube   client.Client
+}
+
+func isUpToDate(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) (bool, error) {
+	if !isUpToDateDeliveryOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateReputationOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateSendingOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateSuppressionOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateTrackingOptions(cr, resp) {
+		return false, nil
+	}
+
+	return svcutils.AreTagsUpToDate(cr.Spec.ForProvider.Tags, resp.Tags)
+}
+
+// isUpToDateDeliveryOptions checks if DeliveryOptions Object are up to date
+func isUpToDateDeliveryOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	if cr.Spec.ForProvider.DeliveryOptions != nil && resp.DeliveryOptions != nil {
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.SendingPoolName) != awsclient.StringValue(resp.DeliveryOptions.SendingPoolName) {
+			return false
+		}
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.TLSPolicy) != awsclient.StringValue(resp.DeliveryOptions.TlsPolicy) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateReputationOptions checks if ReputationOptions Object are up to date
+func isUpToDateReputationOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	if cr.Spec.ForProvider.ReputationOptions != nil && resp.ReputationOptions != nil {
```

Comment:
Is there a case where the `resp.ReputationOptions` is `nil` because user didn't give it in the beginning and then after they update the spec, we need to issue an update in that situation? i.e. should we call in case where `cr.Spec.ForProvider.ReputationOptions != nil && resp.ReputationOptions == nil` is true? Or is `resp.ReputationOptions` always returned with some defaults?

Reply:
Goes for other up-to-date comparisons as well.

Reply:
Check against the AWS API calls
Following options will be returned regardless:
- DeliveryOptions
- ReputationOptions
- SendingOptions

Following options depends on user toggle:
- TrackingOptions
- SuppressionOptions

Will rectify based on these 2 options

Reply:
Anyway, i have rectified these in my previous commits